examples/object_detection_2d/object_detection_2d/seed_test_run.py

```diff
@@ -0,0 +1,160 @@
+from kolena._experimental.object_detection import ThresholdStrategy
+from kolena.workflow.annotation import ScoredLabeledBoundingBox
+from kolena.workflow.test_run import test
+
+
+MODEL_LIST: Dict[str, str] = {
+    "yolo_r": f"YOLOR-D6 (modified CSP, {WORKFLOW})",
+    "yolo_x": f"YOLOX (modified CSP-v5, {WORKFLOW})",
+    "mask_cnn": f"Mask R-CNN (Inception-ResNet-v2, {WORKFLOW})",
+    "faster_rcnn": f"Faster R-CNN (Inception-ResNet-v2, {WORKFLOW})",
+    "yolo_v4s": f"Scaled YOLOv4 (CSP-DarkNet-53, {WORKFLOW})",
+    "yolo_v3": f"YOLOv3 (DarkNet-53, {WORKFLOW})",
+}
+
+# list of all test suites to run if a test suite is not specified
+TEST_SUITE_NAMES = [
+    f"{DATASET} :: transportation brightness [Object Detection]",
+]
+
+
+def model_alias_to_data_path(alias: str) -> str:
+    return S3_MODEL_INFERENCE_PREFIX + alias + "/coco-2014-val_prediction.csv"
+
+
+# transforms test samples into inferences using a dataframe
+def get_stored_inferences(
+    metadata_by_image: pd.DataFrame,
+) -> Callable[[TestSample], Inference]:
+    # a function that creates an inference from a test sample
+    def infer(sample: TestSample) -> Inference:
+        try:
+            image_inferences = metadata_by_image.get_group(sample.locator)
+            return Inference(
+                bboxes=[
+                    ScoredLabeledBoundingBox(
+                        top_left=(float(record.min_x), float(record.min_y)),
+                        bottom_right=(float(record.max_x), float(record.max_y)),
+                        label=str(record.label),
+                        score=float(record.confidence_score),
+                    )
+                    for record in image_inferences.itertuples()
+                ],
+            )
+        except Exception as e:
+            if e is None:
+                print(e)  # ignored
+            return Inference(bboxes=[])
+
+    return infer
+
+
+def seed_test_run(
+    model_alias: str,
+    model_full_name: str,
+    test_suite: TestSuite,
+    groups_df: pd.DataFrame,
+) -> None:
+    # create a model
+    model = Model(model_full_name, infer=get_stored_inferences(groups_df), metadata=MODEL_METADATA[model_alias])
+
+    # customizable configurations for the evaluator
+    evaluator = ObjectDetectionEvaluator(
+        configurations=[
+            ThresholdConfiguration(
+                threshold_strategy=ThresholdStrategy.FIXED_03,
+                iou_threshold=0.3,
+                with_class_level_metrics=True,
+                min_confidence_score=0.2,
+            ),
+            ThresholdConfiguration(
+                threshold_strategy=ThresholdStrategy.F1_OPTIMAL,
+                iou_threshold=0.5,
+                with_class_level_metrics=True,
+                min_confidence_score=0.0,
+            ),
+        ],
+    )
+
+    # runs the evaluation
+    test(model, test_suite, evaluator, reset=True)
+
+
+def main(args: Namespace) -> None:
+    print("loading csv of inferences...")
```

Comment:
This took more than a minute for me to download, during which there was no feedback about what was happening. We should GZip these files to minimize data transfer — the end result will be transparent to the user, and a quick test on `yolo_r` shows the file compressing down to ~60MB from ~330MB.